File: 0400-0499/441/441.cpp
#include <cmath>

using namespace std;

class Solution {
public:
	int arrangeCoins(int n) {
		// See readme for explanation
		// 2n may overflow as int, convert it to long
		long x = sqrt(long(n) * 2);
		return (x * (x + 1) / 2 <= n) ? x : x - 1;
	}
};

File: 0400-0499/441/441.go
package main

import "math"

func arrangeCoins(n int) int {
	// See readme for explanation
	x := int(math.Floor(math.Sqrt(float64(n) * 2)))
	if x*(x+1)/2 > n {
		x = x - 1
	}
	return x
}
